### PR TITLE
log4j2 implemented / public interfaces / used @Param

### PR DESCRIPTION
**Pull request by AndrewTchaikovsky**
1) I applied log4j2 throughout the project. We had both log4j1 and log4j2 in our pom.xml which caused my project on Windows to break. We had both log4j1 and log4j2 dependencies added in Maven. In fact, we only used log4j1 everywhere in the project and maintained a config file for it. I removed the old config, added a new one for log4j2 (it is different now) and re-wrote the methods to work with log4j2 (the methods are also different).
2) I explicitly added "public" to all interfaced as advised by mentor in last class;
3) I added @Param to setWayPoints method.

### CODE REVIEW

**Review by remynewton — APPROVED** [review]
LGTM!

**Review by TomashevskiiVladimirSolvd — APPROVED** [review]
LGTM as well)

File: src/main/resources/mapper/RouteMapper.xml [hunk 1]
```diff
@@ -28,8 +28,8 @@
         SELECT id, start_point, end_point, distance FROM routes
     </select>
 
-    <insert id="setWayPoints">
-        INSERT INTO waypoints (points_id, routes_id) VALUES (#{points_id}, #{routes_id})
+    <insert id="setWayPoints" parameterType="map">
+        INSERT INTO waypoints (points_id, routes_id) VALUES (#{point.id}, #{route.id})
```
Comment (nislam2664) [inline src/main/resources/mapper/RouteMapper.xml:32]: think this should be points.id and routes.id (missing 's') if we're referencing the tables

**Review by kabuki-mono — CHANGES_REQUESTED** [review]
Questionable error on RouteMapper.xml?

**Review by kabuki-mono — APPROVED** [review]
Ignore previous request. I misread the code.

**Pull request merged**